pkg/flare/archive_dca.go

```diff
@@ -176,3 +180,79 @@ func getDCAWorkloadList() ([]byte, error) {
 
 	return getWorkloadList(fmt.Sprintf("https://%v:%v/workload-list?verbose=true", ipcAddress, config.Datadog.GetInt("cluster_agent.cmd_port")))
 }
+
+// getAgentDaemonSet retrieves the DaemonSet manifest of the Agent
+func getAgentDaemonSet(fb flarehelpers.FlareBuilder) error {
+	// The Agent DaemonSet name is based on the Helm chart template and added to the Cluster Agent as an environment variable
+	var agentDaemonsetName string
+	var releaseNamespace string
+	var agentDaemonset []byte
+
+	cl, err := apiserver.GetAPIClient()
+	if err != nil {
+		log.Errorf("Can't create client to query the API Server: %s", err)
+	} else {
+		agentDaemonsetName = os.Getenv(HELM_AGENT_DAEMONSET)
+		releaseNamespace = os.Getenv(HELM_CHART_RELEASE_NAMESPACE)
+		agentDaemonset, err = GetDaemonset(cl, agentDaemonsetName, releaseNamespace)
+		if err != nil {
+			log.Debugf("Error while collecting the Agent DaemonSet: %q", err)
+		}
+	}
+
+	return fb.AddFile("agent-daemonset.yaml", agentDaemonset)
+}
+
+// getClusterAgentDeployment retrieves the Deployment manifest of the Cluster Agent
+func getClusterAgentDeployment(fb flarehelpers.FlareBuilder) error {
+	// The Cluster Agent Deployment name is based on the Helm chart template and added to the Cluster Agent as an environment variable
+	var clusterAgentDeploymentName string
+	var releaseNamespace string
+	var clusterAgentDeployment []byte
+
+	cl, err := apiserver.GetAPIClient()
+	if err != nil {
+		log.Errorf("Can't create client to query the API Server: %s", err)
+	} else {
+		clusterAgentDeploymentName = os.Getenv(HELM_CLUSTER_AGENT_DEPLOYMENT)
+		releaseNamespace = os.Getenv(HELM_CHART_RELEASE_NAMESPACE)
+		clusterAgentDeployment, err = GetDeployment(cl, clusterAgentDeploymentName, releaseNamespace)
+		if err != nil {
+			log.Debugf("Error while collecting the Cluster Agent Deployment: %q", err)
+		}
```

Comment:
I think it it can be safer to check if the envvars results are not empty. 
So maybe return if `clusterAgentDeploymentName` or `releaseNamespace` is empty